### DISCUSSION TITLE
Sudden death + easy mod cross-compatibility

### DISCUSSION THREAD

**Discussion by mk56-spn** [ppy/osu#13857] [Ideas]
https://www.youtube.com/watch?v=d4PfXY7-bkw.

Title plus video really explain it all (yes i also tested without flashlight). I assume this is due to the fact that EZ refills life not being accounted for in the SD mod? 

tested on latest laxer release ( 7th july)

missing on circles also  allows for it

**Comment by bdach** [comment]
I'm not sure fixing this as proposed by forcing easy mod to no extra lives when sudden death is enabled (effectively leading to a situation wherein enabling one mod is affecting settings of another) is a good idea. Sounds like potentially inviting complexity.

Perhaps the extra lives part of easy mod could be split to a separate mod, and that one could be plainly incompatible with sudden death.

**Comment by mk56-spn** [comment]
yeah that makes sense, only reason i brought it up is because right now it feels really not user friendly since the dynamic isnt intuitive, its worth noting that the concept of DT and HT now that rates are a thing also feel out of touch. like it would feel better if it was just a single rate chnage mod with a pitch slider bar  in its options. but thats just me things i guess. not to mention stable users would riot

**Comment by smoogipoo** [comment]
I don't think this is necessarily bad? The bad part for me is that it's not signaled - SD killing you in this situation should probably flash the HP bar or something, along with there being a life count displayed somewhere like a bubble attached to the HP bar.

**Comment by peppy** [comment]
I think the mods being compatible feels fine. Probably needs an audible cue and some kind of visual effect. I think it would potentially be cool to have the hp bar fall away each fail (revealing a new one behind it).

**Comment by Morilli** [comment]
I feel like it makes more sense for the sudden death mod to "override" the extra lives of the easy mod, because
1. It's basically the only thing the sudden death mod does; so it feels like it should be more important than a subfunction of another mod
2. One could argue the main purpose of the sudden death mod is to allow instant-restart on missing a note (or dropping FC), which will no longer be given in this scenario
3. The sudden death mod specifies that it fails a beatmap on miss, not that it drops the health bar to zero. This may be nitpicking but it certainly feels logical this way imo.

**Reply by mk56-spn** [reply]
i kinda agree with morilli, in the sense that having  a mod called sudden death not kill you instantly feels off i guess? maybe if the description is updated to something such as "depletes one hp bar per miss, watch out!" it would make more sense.

**Reply by smoogipoo** [reply]
It does kill you though - you just have multiple lives. The description "miss and fail" would need to be reworded.

**Comment by tovc** [comment]
Bumping this again, even though the behavior isn't going to be changed now. Currently, the Easy mod still gives extra lives that Perfect/Sudden Death respect. The Perfect mod still has the description 'SS or quit.', but you can pass a map without an SS by using Easy. The Sudden Death mod still has the description 'Miss and fail.', but you can miss without failing by using Easy.

Tangential to those mods is the 'Accuracy Challenge' mod, which should do the same thing on paper, but is entirely incompatible with Easy, for some reason. My guess is that it *does* instantly fail you, and instead of making that compatible with Easy's extra lives, the mods were just made entirely incompatible.

The simple solution here would be to change Easy's 'extra lives' option down to 0, but for some reason, that changes the score to unranked, which means it won't be submitted.


My proposal would be to add an 'instant fail' condition, which overrides any extra lives given by easy. Being that that would change the behavior of already submitted scores, it doesn't seem like a realistic option.

Another solution as suggested by bdach, would be to separate the extra lives from Easy into another mod, then retroactively add that to all plays, but that'd still require it be compatible with Sudden Death/Perfect to mirror existing behavior.

I think the simplest solution now would be to simply allow submitting scores with <2 extra lives. I don't know how Accuracy Challenge would be made compatible, but I think that it's silly that it isn't compatible.

**Comment by iiyosu** [comment]
bumping instead of making a new thread.

i also think it makes the most sense for sudden death (and perfect) to override easy's extra lives. i like that they're compatible, but there's not really any reason to play with these mods together because they don't function as they normally do. if it'd cause an issue with preexisting plays to change how these mods interact by default (though i'm not sure why it would), maybe there could be a toggle-able setting that allows this when you pair easy and sd or perfect.
